### ISSUE TITLE
apxs:Error: Command failed with rc=65536

### ISSUE THREAD

**Issue by poljane** [UNINETT/mod_auth_mellon#145]
HI, I'm trying to compile it into an Amazon Linux but it stops compiling and raise up this error: 
```
root@host# make 
/usr/sbin/apxs -Wc,"-std=c99    -DXMLSEC_CRYPTO=\"openssl\" -DXMLSEC_CRYPTO_DYNAMIC_LOADING=1 -D__XMLSEC_FUNCTION__=__FUNCTION__ -DXMLSEC_NO_SIZE_T -DXMLSEC_NO_GOST=1 -DXMLSEC_NO_XKMS=1 -DXMLSEC_DL_LIBLTDL=1 -I/usr/include/libxml2 -I/usr/include/xmlsec1 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include   -DHAVE_LASSO_UTILS_H   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include   -g -O2" -Wl,"-lssl -lcrypto   -llasso   -lcurl   -lglib-2.0  " -Wc,-Wall -Wc,-g -c mod_auth_mellon.c auth_mellon_cache.c auth_mellon_config.c auth_mellon_cookie.c auth_mellon_diagnostics.c auth_mellon_handler.c auth_mellon_util.c auth_mellon_session.c auth_mellon_httpclient.c
/usr/lib64/apr-1/build/libtool --silent --mode=compile gcc -prefer-pic -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -Wformat-security -fno-strict-aliasing  -DLINUX -D_REENTRANT -D_GNU_SOURCE -pthread -I/usr/include/httpd  -I/usr/include/apr-1   -I/usr/include/apr-1  -std=c99    -DXMLSEC_CRYPTO="openssl" -DXMLSEC_CRYPTO_DYNAMIC_LOADING=1 -D__XMLSEC_FUNCTION__=__FUNCTION__ -DXMLSEC_NO_SIZE_T -DXMLSEC_NO_GOST=1 -DXMLSEC_NO_XKMS=1 -DXMLSEC_DL_LIBLTDL=1 -I/usr/include/libxml2 -I/usr/include/xmlsec1 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include   -DHAVE_LASSO_UTILS_H   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include   -g -O2 -Wall -g  -c -o mod_auth_mellon.lo mod_auth_mellon.c && touch mod_auth_mellon.slo
/usr/lib64/apr-1/build/libtool --silent --mode=compile gcc -prefer-pic -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -Wformat-security -fno-strict-aliasing  -DLINUX -D_REENTRANT -D_GNU_SOURCE -pthread -I/usr/include/httpd  -I/usr/include/apr-1   -I/usr/include/apr-1  -std=c99    -DXMLSEC_CRYPTO="openssl" -DXMLSEC_CRYPTO_DYNAMIC_LOADING=1 -D__XMLSEC_FUNCTION__=__FUNCTION__ -DXMLSEC_NO_SIZE_T -DXMLSEC_NO_GOST=1 -DXMLSEC_NO_XKMS=1 -DXMLSEC_DL_LIBLTDL=1 -I/usr/include/libxml2 -I/usr/include/xmlsec1 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include   -DHAVE_LASSO_UTILS_H   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include   -g -O2 -Wall -g  -c -o auth_mellon_cache.lo auth_mellon_cache.c && touch auth_mellon_cache.slo
In file included from auth_mellon_cache.c:22:0:
auth_mellon_cache.c: In function 'am_cache_new':
auth_mellon.h:613:60: error: expected expression before ')' token
 ap_log_rerror(file, line, mi, level, status, r, __VA_ARGS__);
                                                            ^
auth_mellon.h:610:28: note: in expansion of macro 'AM_LOG_RERROR__'
 #define AM_LOG_RERROR(...) AM_LOG_RERROR__(__VA_ARGS__)
                            ^
auth_mellon_cache.c:406:9: note: in expansion of macro 'AM_LOG_RERROR'
         AM_LOG_RERROR(APLOG_MARK, APLOG_ERR, 0, r,
         ^
apxs:Error: Command failed with rc=65536
.
make: *** [mod_auth_mellon.la] Error 1
```

Everybody says that I have to install libtool but it's already installed, anyway it doesn't look like a libtool missing error. 

**Additional info:** 

OS: Amazon Linux 
GCC Version: gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-11)
uname -a : Linux ip-172-31-50-220 4.9.62-21.56.amzn1.x86_64 #1 SMP Thu Nov 16 05:37:08 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

The same version compiled ok in Ubuntu two days ago. 
Some tip? 

Thanks in advance.

**Comment by olavmrk** [comment]
Hi, is this a duplicate of #141? I see you commented on that one afterwards.

**Comment by olavmrk** [comment]
I'm guessing that this is fixed, either as part of #146 or as part of #152. If it isn't, please let me know.

**Comment by mikex2018** [comment]
Hi all,

I am having trouble with this error as well using "master".

Running ./configure does not report any errors, however trying to "make" results in

```
auth_mellon_handler.c: In function ‘am_handle_logout_request’:
auth_mellon_handler.c:708: warning: unused variable ‘cfg’
auth_mellon_handler.c: In function ‘am_handle_logout_response’:
auth_mellon_handler.c:828: warning: unused variable ‘cfg’
auth_mellon_handler.c: In function ‘am_set_authn_request_content’:
auth_mellon_handler.c:2979: error: ‘LASSO_HTTP_METHOD_PAOS’ undeclared (first use in this function)
auth_mellon_handler.c:2979: error: (Each undeclared identifier is reported only once
auth_mellon_handler.c:2979: error: for each function it appears in.)
apxs:Error: Command failed with rc=65536
```

This is on a Lucid box and with the liblasso3 dependencies installed.

**Comment by olavmrk** [comment]
This looks like a different error and a different OS. The only thing that is similar is that apxs fails :smiley:

In any case, considering that Ubuntu 10.04 (Lucid Lynx) was end of life three years ago, I strongly encourage you to upgrade your OS rather than trying to make this work. After all, you are not getting security updates for your OS.
